src/components/widgets/Toggle.tsx

```diff
@@ -0,0 +1,235 @@
+};
+
+/** Fully controlled Toggle component. */
+export default function Toggle({
+  label,
+  labelPosition,
+  styleOverrides,
+  themeRole,
+  state,
+  onToggle,
+  disabled,
+  size = "medium",
+}: ToggleProps) {
+  const theme = useUITheme();
+  const [hoverState, setHoverState] = useState<"default" | "hover">("default");
+
+  const styleSpec: ToggleStyleSpec = useMemo(() => {
+    const themeColor = theme && themeRole && theme.palette[themeRole];
+    const mainColor = themeColor?.hue ?? blue;
+    const mainLevel = themeColor?.level ?? 400;
+    const labelColor = gray[600];
+    const disabledColor = gray[300];
+
+    const defaultStyleSpec: ToggleStyleSpec = {
+      container: {},
+      default: {
+        off: {
+          backgroundColor: "none",
+          knobColor: mainColor[mainLevel],
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+        on: {
+          backgroundColor: mainColor[mainLevel],
+          knobColor: "white",
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+      },
+      hover: {
+        off: {
+          backgroundColor: mainColor[100],
+          knobColor: mainColor[mainLevel],
+          borderColor: mainColor[mainLevel],
+          labelColor,
+        },
+        on: {
+          backgroundColor: mainColor[Math.min(mainLevel + 200, 900)],
+          knobColor: "white",
+          borderColor: mainColor[Math.min(mainLevel + 200, 900)],
+          labelColor,
+        },
+      },
+      disabled: {
+        off: {
+          backgroundColor: "none",
+          knobColor: disabledColor,
+          borderColor: disabledColor,
+          labelColor,
+        },
+        on: {
+          backgroundColor: disabledColor,
+          knobColor: "white",
+          borderColor: disabledColor,
+          labelColor,
+        },
+      },
+    };
+
+    return merge({}, defaultStyleSpec, styleOverrides);
+  }, [styleOverrides, theme, themeRole]);
+
+  /**
+   * The CSS styles that should be applied can depend
+   * on (1) whether or not the component is
+   * focused/hovered, (2) which option is selected,
+   * and (3) whether the toggle is disabled.
+   */
+  const currentStyles = useMemo(() => {
+    const selectedOption = state ? "on" : "off";
+
+    return disabled
+      ? styleSpec.disabled[selectedOption]
+      : styleSpec[hoverState][selectedOption];
+  }, [hoverState, state, styleSpec, disabled]);
+
+  const ariaLabel = useMemo(() => {
+    if (label) return label + " Toggle";
+    else return "Toggle";
+  }, [label]);
+
+  const width = size === "medium" ? 40 : 20;
+  const height = width / 2;
+  const borderWidth = 2;
+  const knobSize = size === "medium" ? 12 : 5;
+  const knobOffset = size === "medium" ? 2 : 1;
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        alignItems: "center",
+        pointerEvents: disabled ? "none" : "auto",
+        fontFamily: primaryFont,
+        fontSize: 13,
```

Comment:
I think Dave made a good point about the `fontSize` of the label for small versus medium toggles. I'm not sure if the best approach is to add an optional prop specifying the label size, or to handle it as a styleOverride.

Reply:
I think given the fact that we're changing the label to be a ReactNode, we can just allow the user to provide their own custom node if they need to customize the label. What do you think?